docs: clarify user journey (#240)
vikejs/telefunc: docs/pages/how-it-works/+Page.mdx

import { Link } from '@brillout/docpress'

You can think of telefunction calls as a type-safe `fetch` call to a registered server-side function.
Under the hood, Telefunc converts telefunction calls into an HTTP request to your server.

Let's see what happens when a telefunction is called.

```js
// hello.telefunc.js
// Environment: server

export { hello }

async function hello({ name }) {
  const message = 'Welcome ' + name
  return { message }
}
```

```js
// Environment: client

import { hello } from './hello.telefunc.js'

const { message } = await hello('Eva')
```

The `hello.telefunc.js` file is never loaded in the browser: instead Telefunc transforms `hello.telefunc.js` into the following:

```js
// hello.telefunc.js (after Telefunc transformation)
// Environment: Browser
import { __internal_makeHttpRequest } from 'telefunc/client'
export const hello = (...args) => __internal_makeHttpRequest('/hello.telefunc.js:hello', args)
```

## Transformer

> Telefunc uses plugins to transform `.telefunc.js` files for:

Reviewer: ```suggestion
Telefunc uses build plugins that transform `.telefunc.js`:
```

Author: i've revised the whole section

> - The client-side, replacing `.telefunc.js` imports with a thin HTTP client.

Reviewer: ```suggestion
- On the client-side, replacing `.telefunc.js` imports with a thin HTTP client.
```

> - The server-side, for <Link href="/shield#automatic-from-typescript" text="automatic TypeScript runtime validation"  

Reviewer: ```suggestion
- On the server-side, for <Link href="/shield#automatic-from-typescript" text="automatic TypeScript runtime validation"  
```

doNotInferSectionTitle={true} />.

> Telefunc plugins also automatically crawl your `.telefunc.js` files (so that you don't have to use <Link href="/telefuncFiles" text={(<code>config.telefuncFiles</code>)} />).

Reviewer: ```suggestion
Telefunc also automatically crawls your `.telefunc.js` files (so that you don't have to use <Link href="/telefuncFiles" text={(<code>config.telefuncFiles</code>)} />).
```
